app/2.0/docs/devguide/dom-template.md

```diff
@@ -44,9 +44,81 @@ except for testing. For caveats about defining elements in the main document,
+if a component includes images alongside an HTML import that defines an element, the image URL needs
+to be resolved relative to the import. However, an application-specific element may need to include
+links to URLs relative to the main document.
+
+To ensure URLs resolve properly, Polymer provides two properties that can be used in data bindings:
+
+| Property | Description |
+| -------- | ----------- |
+| `importPath` | A static getter on the element class that defaults to the element HTML import document URL and is overridable. It may be useful to override `importPath` when an element's template is not retrieved from a `<dom-module>` or the element is not defined using an HTML import. |
+| `rootPath` | An instance property set to the value of `Polymer.rootPath` which is globally settable and defaults to the main document URL. It may be useful to set `Polymer.rootPath` to provide a stable application mount path when using client side routing. |
+
+<b style="color: red;">[[[ Re URLS in CSS styles: Does this include only relative URLs? What if they start
+with / or https://domain/....]]]</b>
+
+URLs in styles are automatically re-written to be relative to the `importPath` property.
+Any URLs outside of a `<style>` element should be bound using `importPath` or
+`rootPath` where appropriate. For example:
+
+```html
+<img src$="[[importPath]]checked.jpg">
+```
+
+```html
+<a href$="[[rootPath]]users/profile">View profile</a>
+```
+
+The `importPath` and `rootPath` properties are also supported in Polymer 1.9+, so they can be used
+by hybrid elements.
+
+### Provide a string template
+
+As an alternative to specifying the element's template in markup, you can specify a string template
+by creating a static `template` getter that returns the template as a string. This getter is called
+_once_, when the first instance of the element is upgraded.
+
+```js
+class MyElement extends Polymer.Element {
+
+  static get template() {
+    return `<style>:host { color: blue; }</style>
+       <h2>String template</h2>
+       <div>I've got a string template!</div>`
+  }
+}
+customElements.define('my-element', MyElement);
+```
+
+When using a string template, the element doesn't need to provide an `is` getter (however, the tag
+name still needs to  be  passed as the first argument to `customElements.define`).
+
+### Elements with no shadow DOM
+
+To create an element with no shadow DOM, don't specify a DOM template (using either a `<dom-module>`
+or a string template), then no shadow root is created for the element, _unless_ the element is
+extending another element that defines a DOM template.
+
+### Inherited templates
+
+An element that extends another Polymer element can inherit its template. If the element doesn't
+provide its own DOM template (using either a `<dom-module>` or a string template), Polymer uses the
+same template as the superclass, if any.
+
+If the superclass has no template, the subclass element doesn't create a shadow root.
```

Comment:
It would be nice in this section to describe how to actually _extend_ templates now that we have inheritance (in addition to this content which just discusses inheriting templates).

So, the way an element gets its template is that there is a default `static get template()` getter that uses the `static get is()` getter if provided to find the template in the dom-module. As described above, one can now override `static get template()` to return an arbitrary template (or string, which is parsed into a template), rather than using `is`+`dom-module`.

So you can now do this kind of thing to augment a super's template if needed:

```js
// clone super's template
let subTemplate = MySuperClass.template.cloneNode(true);
// modify it as desired
subTemplate.appendChild(...);

class MySubClass extends MySuperClass {
  static get template() { return subTemplate; }
}
```